### PR TITLE
add keyword lookup by PowerShell Get-Help

### PR DESCRIPTION
**Pull request by Konfekt**
To look up keywords by PowerShell's Get-Help, hit the `K` key.

### CODE REVIEW

**Comment by Konfekt** [comment]
This was also proposed at the fork https://github.com/zigford/vim-powershell/pull/

File: doc/ps1.txt [hunk 1]
```diff
@@ -50,5 +50,9 @@ To configure whether to show the exception type information: >
     :let g:ps1_efm_show_error_categories = 1
 <
 
+KEYWORD LOOKUP                                                 *ps1-keyword*
+
+To look up keywords by PowerShell's Get-Help, hit the |K| key.
```
Comment (heaths) [inline doc/ps1.txt]: "by" -> "using"

**Comment by heaths** [comment]
Actually, you might also pop back up to the top of the buffer. Trying this out, the cursor is positioned at the bottom of the buffer.

**Comment by Konfekt** [comment]
> Actually, you might also pop back up to the top of the buffer

I think you are opening a fairly recent version of Vim with terminal support in a GUI or without less installed (such as https://chocolatey.org/packages/less on Windows). In this case, the documentation is shown and the cursor at the end of the output. I do not know an easy fix to it.

**Comment by heaths** [comment]
*Requiring* extra stuff to be installed is not permitted. You can execute `cursor(1, 1)` to position the cursor without piping through less. In fact, if that's all less is being used for, please remove the call (will be faster than exec'ing another process).

I'm using vim.exe in Windows Terminal, but same happens in pwsh.exe using conhost (Windows' default console host).

**Comment by Konfekt** [comment]
> Requiring extra stuff to be installed is not permitted.

It is not required, but the pager less is more convenient than the built-in Vim pager, for examplee, it does not permit to scroll it by the built-in pager, as amply asked for, see for example,

https://stackoverflow.com/a/49146042
https://stackoverflow.com/a/7831417
https://stackoverflow.com/q/5693959 or
https://stackoverflow.com/a/19776493

Bram responded that the pager is a script, thus anyone can improve it. Therefore, a pager such as `less` (which was opted for because it is the most ubiquitous) is meant to be used with it.

> You can execute cursor(1, 1) to position the cursor without piping through less

Even simpler, type `gg` in normal mode to go to the first line.

> I'm using vim.exe in Windows Terminal, but same happens in pwsh.exe using conhost (Windows' default console host).

If you are using Vim in a Terminal, then perhaps `less` is not installed?

**Comment by heaths** [comment]
Adding `cursor(1,1)` obviates having to install anything extra. It's a one line change and blocking merge.

**Comment by Konfekt** [comment]
For example,

```vim
command! -buffer -nargs=1 GetHelp silent exe (has('terminal') ? 'term ' : '!') . s:pwsh_cmd . ' -NoLogo -NoProfile -NonInteractive -ExecutionPolicy RemoteSigned -Command Get-Help -Full' | call cursor(1,1)
setlocal keywordprg=:GetHelp
```

does not change anything. 

The terminal buffer has to be filled first before `call cursor()` has an effect. Therefore,  a three seconds delay was added. However, my machine is over a decade old. Perhaps `call cursor()` alone on a recent machine is entirely sufficient.

**Comment by heaths** [comment]
1. Shouldn't that be added to the `else` condition as well?
2. After the recent changes, it's not working for me anymore. `K` reports the error "Mark not set". It was working previously.

**Comment by Konfekt** [comment]
> Shouldn't that be added to the else condition as well?

That refers to `call cursor()` ? This only works in a buffer, but not in Vim's built-in pager which we only use it everything else fails. In fact, already `system(...)` is a hack.

> error "Mark not set"
I think the pushed version was missing an `exe`. Please try again.

**Comment by heaths** [comment]
This worked for me:

```diff
diff --git a/ftplugin/ps1.vim b/ftplugin/ps1.vim
index 3339768..89ad5c8 100644
--- a/ftplugin/ps1.vim
+++ b/ftplugin/ps1.vim
@@ -43,7 +43,7 @@ if exists('s:pwsh_cmd')
   elseif has('terminal')
     command! -buffer -nargs=1 GetHelp silent exe 'term ' . s:pwsh_cmd . ' -NoLogo -NoProfile -NonInteractive -ExecutionPolicy RemoteSigned -Command Get-Help -Full <args>' | call term_wait('', 3000) | call cursor(1,1)
   else
-    command! -buffer -nargs=1 GetHelp echo system(s:pwsh_cmd . ' -NoLogo -NoProfile -NonInteractive -ExecutionPolicy RemoteSigned -Command Get-Help -Full <args>')
+    command! -buffer -nargs=1 GetHelp echo system(s:pwsh_cmd . ' -NoLogo -NoProfile -NonInteractive -ExecutionPolicy RemoteSigned -Command Get-Help -Full <args>') | call cursor(1,1)
   endif
 endif
 setlocal keywordprg=:GetHelp
```

**Comment by Konfekt** [comment]
This is strange, as your Vim probably has terminal support judging from your first message and therefore the last change should not affect the output. Would you mind checking the latest forceful push?

**Comment by heaths** [comment]
Both gvim and vim are compiled with 'terminal'. I was playing around with this some more and can't get something to work stably. Seems there's no reliable way to do this, and nothing I can find specific to `keywordprg` either (since it manages the buffer itself). It's not ideal, but I'd just get rid of the `term_wait` and `cursor` calls. If they won't work reliably (e.g. longer help output affect the timing) there's no point in having them and an inconsistent experience. Users will just have to `gg`.

**Comment by Konfekt** [comment]
Yes, the time necessary to print the help depends on the keyword looked up. Sometimes a second suffices, sometimes not even five. Everything is fine once `less` installed. Perhaps one could recommend its installation as referred to above?

**Pull request merged**